### PR TITLE
OHRM5X-1359: Candidate Workflow

### PR DESCRIPTION
**Pull request by SandeepaOHRM**
This PR contains:-

  Entities:-
  - CandidateHistory
  - Interview
  - InterviewAttachment
  
API to get allowed candidate actions

### CODE REVIEW

File: src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/CandidateHistoryDecorator.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    /**
+     * @param int $id
+     */
+    public function setVacancyById(int $id): void
+    {
+        $vacancy = $this->getReference(Vacancy::class, $id);
+        $this->candidateHistory->setVacancy($vacancy);
+    }
+
+    public function setInterview(int $id): void
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/CandidateHistoryDecorator.php:63]: doc comment

File: src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/InterviewAttachmentDecorator.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        if (is_null($this->attachmentString) && is_resource($fileContent)) {
+            $this->attachmentString = stream_get_contents($fileContent);
+        }
+        return $this->attachmentString;
+    }
+
+    /**
+     * @return InterviewAttachment
+     */
+    public function getInterviewAttachment(): InterviewAttachment
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/InterviewAttachmentDecorator.php:66]:
```suggestion
    protected function getInterviewAttachment(): InterviewAttachment
```

File: src/plugins/orangehrmRecruitmentPlugin/entity/Interview.php [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+namespace OrangeHRM\Entity;
+
+use DateTime;
+use Doctrine\Common\Collections\ArrayCollection;
+use OrangeHRM\Entity\Decorator\DecoratorTrait;
+use OrangeHRM\Entity\Decorator\InterviewDecorator;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @method InterviewDecorator getDecorator()
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/entity/Interview.php:29]:
```suggestion
 * @method InterviewDecorator getDecorator()
 *
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/CandidateHistoryDecorator.php (hunk 1 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/CandidateHistoryDecorator.php:63]: done

File: src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/InterviewAttachmentDecorator.php (hunk 2 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/entity/Decorator/InterviewAttachmentDecorator.php:66]: done

File: src/plugins/orangehrmRecruitmentPlugin/entity/Interview.php (hunk 3 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/entity/Interview.php:29]: done

File: src/plugins/orangehrmRecruitmentPlugin/entity/CandidateHistory.php [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+     * @var string|null
+     * @ORM\Column(name="note", type="text", nullable=true)
+     */
+    private ?string $note = null;
+
+    /**
+     * @var string|null
+     * @ORM\Column(name="interviewers", type="text", nullable=true)
+     */
+    private ?string $interviewers = null;
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/entity/CandidateHistory.php:104]: is this correct?

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**